### PR TITLE
Create Genre VC and Reuse Top VC

### PR DESCRIPTION
**Pull request by sheep-q**
## [Pull request title]
 - Create Genre VC
 - Fetch Data into Genre VC
 - Reuse Top VC 
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/projects/ios-play/issues?sort=id
#### Screenshots (nếu có)
Screen Shot 2021-04-19 at 12 58 07 PM

### CODE REVIEW

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 1]
```diff
@@ -203,6 +210,29 @@ extension AppViewController : UICollectionViewDelegate, UICollectionViewDataSour
...
+            }
+        }  else if nextIndex == 5 {
+            if let vc = childView[nextIndex] as? TopAppViewController {
+                vc.urlString = AppUrl.grossingIpad.url
+            }
+        }  else if nextIndex == 6 {
+            if let vc = childView[nextIndex] as? TopAppViewController {
+                vc.urlString = AppUrl.topPaid.url
+            }
+        }
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]: dùng enum, không if else như này

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 2]
```diff
@@ -49,6 +49,13 @@ final class AppViewController: UIViewController {
     
     private var currentPageIndex = 0
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        navigationController?.do {
+            $0.delegate = self
+        }
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]: navigation controller nên implement delegate bên trong base navigation

File: iOS-Play/iOS-Play/Controller/BookViewController/BookViewController.swift [hunk 3]
```diff
@@ -29,6 +29,13 @@ final class BookViewController: UIViewController {
     
     private var currentPageIndex : Int = 0
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        navigationController?.do {
+            $0.delegate = self
+        }
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/BookViewController/BookViewController.swift]: set delegate bằng chính navigationcontroller bên trong BaseNavigationController

File: iOS-Play/iOS-Play/Controller/Genre VC/GenreTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import UIKit
+import Reusable
+
+class GenreTableViewCell: UITableViewCell, NibReusable {
+
+    @IBOutlet weak var appImage: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var releaseDateLabel: UILabel!
+    override func awakeFromNib() {
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/Genre VC/GenreTableViewCell.swift]:
```suggestion
    @IBOutlet weak var releaseDateLabel: UILabel!
    
    override func awakeFromNib() {
```

**Review by dongtg-1485 — APPROVED** [review]

File: iOS-Play/iOS-Play/Controller/AppViewController/Top App/TopAppViewController.swift [hunk 5]
```diff
@@ -23,13 +23,17 @@ final class TopAppViewController: UIViewController {
     
     private let apiManager = APIManager.shared
     
+    var urlString = ""
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/AppViewController/Top App/TopAppViewController.swift]:
```suggestion
        super.viewWillAppear(animated)
```

File: iOS-Play/iOS-Play/Controller/Genre VC/GenreTableViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+//
+
+import UIKit
+import Reusable
+
+class GenreTableViewCell: UITableViewCell, NibReusable {
+
+    @IBOutlet weak var appImage: UIImageView!
+    @IBOutlet weak var nameLabel: UILabel!
+    @IBOutlet weak var releaseDateLabel: UILabel!
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Genre VC/GenreTableViewCell.swift]: private

File: iOS-Play/iOS-Play/Controller/Genre VC/GenreViewController.swift [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+//  iOS-Play
+//
+//  Created by Quang Nguyen on 4/17/21.
+//
+
+import UIKit
+
+class GenreViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Genre VC/GenreViewController.swift]: private

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**